contracts/exchange-libs/contracts/src/LibTransactionDecoder.sol

```diff
@@ -0,0 +1,191 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.5;
+pragma experimental "ABIEncoderV2";
+
+import "@0x/contracts-exchange-libs/contracts/src/LibExchangeSelectors.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+
+
+contract LibTransactionDecoder is LibExchangeSelectors {
+    using LibBytes for bytes;
+
+    /// @dev Decodes the call data for an Exchange contract method call.
+    /// @param signedTransactionData ABI-encoded calldata for an Exchange
+    ///     contract method call.
+    /// @return The name of the function called, and the parameters it was
+    ///     given.  For single-order fills and cancels, the arrays will have
+    ///     just one element.
+    function decodeTransaction(bytes memory signedTransactionData)
+        public
+        view
```

Comment:
Can't this be `pure`?

Reply:
I swore I tried it as `pure` and hit a problem because of accessing the selector constants, but I just changed it to `pure` locally and it built and tested just fine.  I'll change it.